carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/decorator/ExtendedFieldDecorator.java

```diff
@@ -165,6 +173,8 @@ protected ExtendedWebElement proxyForLocator(ClassLoader loader, Field field, El
     @SuppressWarnings("unchecked")
     protected <T extends AbstractUIObject> List<T> proxyForListUIObjects(ClassLoader loader, Field field,
             ElementLocator locator) {
+    	LOGGER.debug("Setting setShouldCache=false for locator: " + getLocatorBy(locator).toString());
+    	((ExtendedElementLocator) locator).setShouldCache(false);
         InvocationHandler handler = new AbstractUIObjectListHandler<T>((Class<?>) getListType(field), webDriver,
                 locator, field.getName());
         List<T> proxies = (List<T>) Proxy.newProxyInstance(loader, new Class[] { List.class }, handler);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "proxies". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://52.53.207.202/sonarqube/coding_rules#rule_key=squid%3AS1488)